Modules/Core/Common/test/itkPointSetGTest.cxx

```diff
@@ -80,3 +81,35 @@ TEST(PointSet, SetPointsByCoordinates)
   TestSetPointsByCoordinates(*itk::PointSet<int>::New());
   TestSetPointsByCoordinates(*itk::PointSet<double, 2, itk::QuadEdgeMeshTraits<double, 2, bool, bool>>::New());
 }
+
+
+// Tests that `PointSet::Graft` just copies the *pointers* to the points and the data. It does not do a "deep copy".
+TEST(PointSet, GraftDoesShallowCopy)
+{
+  const auto check = [](const auto & pointSet) {
+    const auto clone = pointSet.Clone();
```

Comment:
This tests not only that Graft does not do a deep copy, but that Clone does not do a deep copy either.

Reply:
Indeed @dzenanz Note that `pointSet.Clone()` does not even do a _shallow_ copy! It just creates an empty `PointSet` that has the same `PixelType` and `PointType` as the original. Is that OK?

Obviously the test just checks the behavior that `PointSet::Graft` always had. It seems useful to me to support deep copying, but adjusting `PointSet::Graft` to do so would possibly break legacy code. 🤷

Reply:
FYI, I'm considering to add deep copy functionality to PointSet and PointSetBase:
https://github.com/InsightSoftwareConsortium/ITK/compare/master...N-Dekker:ITK:PointSetBase-Clone-MakeDeepCopy